packages/plugin-react-native-span-access/android/src/main/java/com/bugsnag/reactnative/performance/nativespans/JavascriptSpanTransactionImpl.java

```diff
@@ -0,0 +1,151 @@
+package com.bugsnag.reactnative.performance.nativespans;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.WritableArray;
+import com.facebook.react.bridge.WritableMap;
+
+import java.util.LinkedHashMap;
+import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+class JavascriptSpanTransactionImpl implements JavascriptSpanTransaction {
+
+  private static final String ID = "id";
+  private static final String NAME = "name";
+  private static final String VALUE = "value";
+  private static final String ATTRIBUTES = "attributes";
+  private static final String END_TIME = "endTime";
+  private static final String IS_ENDED = "isEnded";
+
+  private final String spanName;
+
+  private final LinkedHashMap<String, Object> attributes = new LinkedHashMap<>();
+
+  private boolean isEnded = false;
+  private long endTime = -1L;
+  private boolean isOpen = true;
+
+  JavascriptSpanTransactionImpl(String spanName) {
+    this.spanName = spanName;
+  }
+
+  @Override
+  public JavascriptSpanTransaction end() {
+    return end(-1L);
+  }
+
+  @Override
+  public JavascriptSpanTransaction end(long endTime) {
+    if (!isOpen) {
+      return this;
+    }
+
+    this.isEnded = true;
+    this.endTime = endTime >= 0
+            ? BugsnagClock.INSTANCE.unixNanoTimeToElapsedRealtime(endTime)
```

Comment:
I think we want to convert _to_ unix nanoseconds here:
```suggestion
            ? BugsnagClock.INSTANCE.elapsedNanosToUnixTime(endTime)
```